### DISCUSSION TITLE
Installation error on Ubuntu

### DISCUSSION THREAD

**Discussion by sitangshugk95** [ContactEngineering/SurfaceTopography#239] [Q&A] [answered]
I am trying to install on Ubuntu 20.04.5 LTS, and I am getting the following errors:
[errors.txt](https://github.com/ContactEngineering/SurfaceTopography/files/9942344/errors.txt)
Could you please help me with this?

**Comment by pastewka** [comment]
Hi @sitangshugk95 - you need to install the FFTW headers, `libfftw3-dev`, i.e. run `sudo apt install libfftw3-dev` before attempting to install `SurfaceTopography`.

**Reply by sitangshugk95** [reply]
Hello @pastewka , 
I do not have sudo access on my university system, but I have an option to install packages for standard repositories using self-service. 
I went through this [singularity](https://github.com/ContactEngineering/SurfaceTopography/blob/master/singularity/SurfaceTopography_serial.def) that you wrote, and I have installed libfftw3-dev, libnetcdf-dev and libopenblas-dev. I have also run `apt list --installed` and here is the list which shows all packages installed (that also shows libfftw3 installed, line 890) : 
[installedpackages.txt](https://github.com/ContactEngineering/SurfaceTopography/files/9943267/installedpackages.txt) .
Any other suggestions as to what might be causing this?

**Reply by pastewka** [reply]
Do you mean the Singularity recipe does not work? You cluster should have FFTW3 headers installed somewhere, at least I'd be surprised if they were missing. In any case, you don't need them on the cluster if you deploy using Singularity Containers.

**Reply by sitangshugk95** [reply]
I meant to say that I have managed to install the libfftw3-dev package.
Is there a command to check if it has been installed correctly? I cannot understand what the error (attached in the initial message) means? Does it need to be in a certain folder, that maybe I can move it to?

**Reply by sitangshugk95** [reply]
I guess I don't quite understand what "deploying using Singularity Container" means - I'm using a lab PC (and not a cluster). 

Should I just copy-paste the lines written in this https://github.com/ContactEngineering/SurfaceTopography/blob/master/singularity/SurfaceTopography_serial.def ?

Or are these just the steps mentioned, and I still need to find the necessary commands to install each of these using pip install?

**Comment by pastewka** [comment]
@sannant What is the status of the muFFT stub?

**Comment by pastewka** [comment]
@sannant Can you help @sitangshugk95 here?

**Reply by sitangshugk95** [reply]
@sannant if you want me to give a step-by-step screenshot/txt file of each step I have followed, I can do that too. 

Firstly, which one of these 2 approaches should I follow for installation - 

1. Write each line on the Ubuntu terminal, as mentioned the installation documentation?
2. Use this singularity container https://github.com/ContactEngineering/SurfaceTopography/blob/master/singularity/SurfaceTopography_serial.def instead? (I have no idea how to use it, and if it is meant to be an alternative)?

**Comment by sitangshugk95** [comment, marked as answer]
@pastewka @sannant 


> I am trying to install on Ubuntu 20.04.5 LTS, and I am getting the following errors: [errors.txt](https://github.com/ContactEngineering/SurfaceTopography/files/9942344/errors.txt) Could you please help me with this?

This issue was solved by uninstalling numpy and then reinstalling using `pip install numpy --no-binary numpy` . Surface Topography and Contact Mechanics were successfully installed after this.
